sphinx/screens-detail/subscription/subscription/src/main/java/chat/sphinx/subscription/ui/SubscriptionViewModel.kt

```diff
@@ -39,23 +41,40 @@ internal class SubscriptionViewModel @Inject constructor(
 {
     private val args: SubscriptionFragmentArgs by savedStateHandle.navArgs()
 
+    companion object {
+        const val DAILY_INTERVAL: String = "daily"
+        const val WEEKLY_INTERVAL: String = "weekly"
+        const val MONTHLY_INTERVAL: String = "MONTHLY"
+    }
+
     fun initSubscription() {
         viewModelScope.launch(mainImmediate) {
-            subscriptionRepository.getActiveSubscriptionByContactId(ContactId(args.argContactId)).firstOrNull().let { subscription ->
+            subscriptionRepository.getActiveSubscriptionByContactId(ContactId(args.argContactId)).distinctUntilChanged().collect { subscription ->
```

Comment:
this change now results in a never ending collection of the subscription that is initialized every time the fragment view goes through the `onViewCreated()` resulting in multiple coroutines collecting. Needs to be moved to a custom view state container which utilizes a flow builder and `.stateIn()` extension function with WhileSubscribed in order to cancel sub when screen's lifecycle moves to the `onStop` event.